jquery.xblock.js

```diff
@@ -348,4 +377,19 @@
         });
     };
 
+    // Find all the children of an element that match the selector, but only
+    // the first instance found down any path.  For example, we'll find all
+    // the ".xblock" elements below us, but not the ones that are themselves
+    // contained somewhere inside ".xblock" elements.
+    // Code borrowed from edx-platform and edx-sdk
+    $.fn.highestDescendants = function(selector) {
+        return this.children().map(function(idx, element) {
+            if ($(element).is(selector)) {
+                return element;
+            } else {
+                return $(element).highestDescendants(selector).toArray();
+            }
```

Comment:
@e-kolpakov I know, but that's actually desired behaviour. [The jQuery `map` function](https://api.jquery.com/jquery.map/) creates an array from all the results, and it will flatten any returned arrays and mix them in with single elements that are returned.

The code above is also found in edx-platform and edx-sdk.